### PR TITLE
Add grammar file for Coffeescript

### PR DESCRIPTION
**Pull request by ClasherKasten**

### CODE REVIEW

**Comment by asottile** [comment]
are people still using coffescript? I thought it was dead

**Comment by ClasherKasten** [comment]
[devdocs](https://github.com/freeCodeCamp) still using it, though there is work to migrate it ([PR 1441](https://github.com/freeCodeCamp/devdocs/pull/1441)) it is unsure when these changes will be complete and merged.
Also the language color highlights `.cson` files, which could be somewhat beneficial for working on the grammars this project relies on that are oft this type.
Example from `atom-language-rust`:
- with the proposed grammar
Screenshot from 2023-08-03 15-19-40
- without the proposed grammar
Screenshot from 2023-08-03 15-19-49

File: bin/download-syntax [hunk 1]
```diff
@@ -280,6 +280,15 @@ REPOS = (
         license_path='LICENSE.txt',
         grammars=('syntaxes/SQL.plist',),
     ),
+    Repo(
+        name='microsoft/vscode-textmate',
+        version='975e2e3',
+        license_path='LICENSE.md',
+        grammars=(
+            'test-cases/themes/syntaxes/coffeescript.json',
+            'test-cases/themes/syntaxes/Regular Expressions (JavaScript).tmLanguage',  # noqa: E501
```
Comment (asottile) [inline bin/download-syntax:289]: hmmm I don't think these are real grammars since they're pulled out of the tests of the textmate engine

Comment (asottile) [reply bin/download-syntax:289]: the test file had to have come from _somewhere_ though -- I'm not really comfortable using a file from tests

Comment (ClasherKasten) [reply bin/download-syntax:289]: There is a grammar in [microsoft/vscode](https://github.com/microsoft/vscode/blob/main/extensions/coffeescript/syntaxes/coffeescript.tmLanguage.json), though the base for this is an atom project which is dead. Another alternative could be forking and updating (textmate/coffee-script.tmbundle)[https://github.com/textmate/coffee-script.tmbundle] as this project is dead too and doesn't seem to provide as much rules as the vscode one.

Comment (asottile) [reply bin/download-syntax:289]: we pull some other ones directly from vscode -- this one is probably fine too

**Review by asottile — APPROVED** [review]

**Pull request merged**